tests/jballerina-unit-test/src/test/resources/test-src/types/readonly/test_selectively_immutable_type.bal

```diff
@@ -192,18 +196,22 @@ function testSimpleInitializationForSelectivelyImmutableMappingTypes() {
     assertTrue(r3 is Student & readonly);
 
     val = r3;
-    Student stVal = <Student> val;
-    assertTrue(stVal.isReadOnly());
-    assertTrue(stVal.details.isReadOnly());
-    assertEquality(<Details> {name: "Jo", id: 4567}, stVal.details);
-
-    assertTrue(stVal["math"] is [RESULT, int] & readonly);
-    assertTrue(stVal["math"].isReadOnly());
-    assertEquality(<[RESULT, int]> ["P", 75], stVal["math"]);
-
-    assertTrue(stVal["science"] is [RESULT, int] & readonly);
-    assertTrue(stVal["science"].isReadOnly());
-    assertEquality(<[RESULT, int]> ["P", 65], stVal["science"]);
+    if (val is error) {
```

Comment:
I think we can do 

```
assertFalse(val is error);
```

and move the logic in the else block also out.

Reply:
Applicable in other places too.

Reply:
I changed val to `any|error` type. Because of this cannot cast out error from this right?
incompatible types: '(any|error)' cannot be cast to 'Employee'
incompatible types: '(any|error)' cannot be cast to 'Student'

Reply:
We can do `Employee rec = <Employee> checkpanic val;`